modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/utils/actionItems/filterItemActions.ts

```diff
@@ -22,12 +22,20 @@ const filterItemActions = (
 							)
 						) {
 							if (action.target === 'headless') {
+								const matchedPermissionKey = Object.keys(
+									itemData.actions
+								).filter(
+									(itemAction) =>
+										itemAction.toLowerCase() ===
+										action.data?.permissionKey?.toLowerCase()
+								);
+
 								return [
 									...actions,
 									{
 										...action,
 										...itemData.actions[
-											action.data.permissionKey.toLowerCase()
+											matchedPermissionKey[0]
```

Comment:
Could this technically evaluate to `...itemData.actions[undefined]`. Would be more fail safe to only use `...itemData.actions[matchedPermissionKey[0]]` if `matchedPermissionKey[0]` is not undefined

Reply:
Based on condition of [L20](https://github.com/liferay-frontend/liferay-portal/pull/4101/files#diff-5c1469eeb87183817f43a118758d7fc81e744f63a65fc3dc5b4120681e8f4403R20), it cannot evaluate to `undefined`.

```
if (action.data?.permissionKey) {
	if (
		itemData.actions &&
		Object.keys(itemData.actions).some(
			(itemAction) =>

                               // if the action name does not match with permissionKey will return false

				itemAction.toLowerCase() ===
				action.data?.permissionKey?.toLowerCase()
		)
```